### PR TITLE
Add optional support for http mode (w/o TLS)

### PR DESCRIPTION
**Pull request by naiquevin**
The default continues to be https. Http mode is primarily meant for local development. But it could also be useful if the db is behind a reverse proxy that takes care of SSL. 

Besides the main change, this PR also includes a commit to load TLS config (paths to the certificate files) from `config.toml` file instead of an env var.

### CODE REVIEW

**Comment by tinkn** [comment]
It looks good, could we use strict typing for host & port please?

```
use std::net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr};

// Custom type for Host
#[derive(Debug, Clone)]
enum Host {
    Ip(IpAddr),
    Hostname(String),
}

// Custom type for Port
#[derive(Debug, Clone, Copy)]
struct Port(u16);

// Implement From trait for convenient conversion
impl From<u16> for Port {
    fn from(port: u16) -> Self {
        Port(port)
    }
}
```

File: src/web_server.rs [hunk 1]
```diff
@@ -116,42 +128,35 @@ pub async fn run_actix_server() -> std::io::Result<()> {
...
         // .service(web::resource("/manual").route(web::post().to(index_manual)))
         // .service(web::resource("/").route(web::post().to(index)))
-    })
-    .bind_rustls_0_23(format!("{}:{}", config_data.server.host, config_data.server.port), config)?
-    .run()
-    .await
+    });
+
+    let port = u16::from(config_data.server.port);
+    let server = match (tls, &config_data.server.host) {
```
Comment (naiquevin) [inline src/web_server.rs:134]: I am not very happy with this part. But implementing `std::net::ToSocketAddrs` trait for `Host` doesn't seem to be trivial as the `ToSocketAddrs` impl for `IpAddr` v/s `String` return different types https://doc.rust-lang.org/std/net/trait.ToSocketAddrs.html#implementors

**Comment by tinkn** [comment]
Not sure am understanding the issue right, but this works

```
use std::net::{IpAddr, SocketAddr, ToSocketAddrs};
use std::io;
use std::net::{Ipv4Addr, Ipv6Addr};

// Custom type for Host
#[derive(Debug, Clone)]
enum Host {
    Ip(IpAddr),
    Hostname(String),
}

// Custom type for Port
#[derive(Debug, Clone, Copy)]
struct Port(u16);

// Custom type that wraps Host and Port
struct HostPort(Host, Port);

// Implement From trait for convenient conversion
impl From<u16> for Port {
    fn from(port: u16) -> Self {
        Port(port)
    }
}

// Implement ToSocketAddrs for the HostPort type
impl ToSocketAddrs for HostPort {
    type Iter = std::vec::IntoIter<SocketAddr>;

    fn to_socket_addrs(&self) -> io::Result<Self::Iter> {
        match &self.0 {
            Host::Ip(ip) => {
                let socket = SocketAddr::new(*ip, self.1.0);
                Ok(vec![socket].into_iter())
            }
            Host::Hostname(hostname) => {
                let addresses = (hostname.as_str(), self.1.0).to_socket_addrs()?;
                Ok(addresses.collect::<Vec<_>>().into_iter())
            }
        }
    }
}

fn main() {
    // Example usage
    let host = Host::Ip(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)));
    let port = Port(8080);
    
    // Convert to SocketAddr using HostPort
    let host_port = HostPort(host, port);
    let socket_addrs: Vec<SocketAddr> = host_port.to_socket_addrs().unwrap().collect();
    
    println!("{:?}", socket_addrs);
}
```

**Comment by naiquevin** [comment]
Updated the commit integrating the above code. Please check.

**Pull request merged**